src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1306,16 +1318,35 @@ private static void ComputeDeclarationsInNode(SemanticModel semanticModel, ISymb
                         executableCodeBlocks = declInNode.ExecutableCodeBlocks;
 
                         // Execute operation actions.
-                        if (shouldExecuteOperationActions && executableCodeBlocks.Any())
+                        if ((shouldExecuteOperationActions || shouldExecuteOperationBlockActions) && executableCodeBlocks.Any())
```

Comment:
You can probably extract out the code within this if statement into a separate method - should make it easier to read.